var/spack/repos/builtin/packages/apptainer/package.py

```diff
@@ -64,3 +70,25 @@ def flag_handler(self, name, flags):
         # Certain go modules this build pulls in cannot be built with anything
         # other than -O0. Best to just discard any injected flags.
         return (None, flags, None)
+
+    # They started vendoring the fuse bits and assume they'll be in the
+    # libexec/apptainer prefix as a result. When singularity is run with
+    # suid it doesn't search the user's $PATH for security reasons.
+    # Since we don't use the vendored deps and instead install them in
+    # their own prefixes they are not found by default.
+    # This is likely only relevant for 1.3:, but it should be fine everywhere
+    @run_after("install")
+    def fix_binary_path(self):
+        paths = []
+        # default value
+        binpath = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+        for i in [
+            s for s in ["e2fsprogs", "gocryptfs", "squashfuse", "fuse-overlayfs"] if s in self.spec
+        ]:
+            paths.append(self.spec[i].prefix.bin)
```

Comment:
Do we want to prefer spack packages in case of local installs?
```suggestion
            paths.prepend(self.spec[i].prefix.bin)
```

Reply:
we do, binpath (the default search path per apptainer docs)  is appended to paths the line below. The net result should be $PATH:the:spack:paths:defaultbinpath